### DISCUSSION TITLE
Is there a way to add a new function to the right-click file context menu?

### DISCUSSION THREAD

**Discussion by JackYinpei** [jupyterlite/jupyterlite#1218] [Q&A]
Sorry, I opened an issue by mistake.

Hello, I have been using jupyterlite-core recently to deploy Jupyter on my local machine. 

####Now I want to add a new feature where users can right-click on a file and save it to the cloud. 
**right here**
![image](https://github.com/jupyterlite/jupyterlite/assets/49555788/5ca3bb0b-6a57-4aa2-bd51-ae1f5ec94053)

I tried implementing this feature using extensions, but that requires modifying JupyterLab. However, if I modify JupyterLab, how can I combine it with JupyterLite? I haven't seen any articles related to this.

What I really want to know is how to combine JupyterLite with my own modified version of JupyterLab for use.🤣🤣🤣🤣

 Any information you can provide would be appreciated. Thank you in advance.

**Comment by jtpio** [comment]
> Sorry, I opened an issue by mistake.

No problem, just closed https://github.com/jupyterlite/jupyterlite/issues/1217 in favor of this discussion.

> I tried implementing this feature using extensions, but that requires modifying JupyterLab.

Normally it should not require modifying JupyterLab. And making an extension would be the preferred way as it would also work in JupyterLite. What part of JupyterLab did you have to modify?

It is possible to disable core extensions and provide different implementations or fonctionality via third-party extensions, including new context menu items.

Maybe having a look at this example would be a first good step?  https://github.com/jupyterlab/extension-examples/tree/main/context-menu

**Reply by JackYinpei** [reply]
> What part of JupyterLab did you have to modify?

After reviewing the content in the links you provided above, I think there is no need to modify JupyterLab for now.

> It is possible to disable core extensions 

Is it to disable the core extensions through the method described in the content of the link below?
[https://jupyterlite.readthedocs.io/en/stable/howto/configure/advanced/extensions.html#disabling-extensions-at-runtime](url)

> provide different implementations or fonctionality via third-party extensions

After I add a new feature, I can then load the plugin into JupyterLite using the method described in the link below.
[https://jupyterlite.readthedocs.io/en/stable/howto/extensions/frontend.html#load-the-extension-in-jupyterlite](url)

So kind of you!!! thanks very much
